demo/make_local_dataset_app.py

```diff
@@ -0,0 +1,214 @@
+        If None, start from 0 or append after the last element in the existing .csv
+
+    Returns:
+        str: Status text for Gradio app
+    """
+    Path(dataset_dir).mkdir(parents=True, exist_ok=True)
+    if Path(f"{dataset_dir}/text.csv").is_file():
+        text_df = pd.read_csv(f"{dataset_dir}/text.csv")
+        if index is None:
+            index = len(text_df)
+        text_df = pd.concat(
+            [text_df, pd.DataFrame([{"index": index, "sentence": sentence}])],
+            ignore_index=True,
+        )
+        text_df = text_df.drop_duplicates().reset_index(drop=True)
+    else:
+        if index is None:
+            index = 0
+        text_df = pd.DataFrame({"index": index, "sentence": [sentence]})
+
+    text_df = text_df.sort_values(by="index")
+    text_df.to_csv(f"{dataset_dir}/text.csv", index=False)
+
+    audio_filepath = f"{dataset_dir}/rec_{index}.wav"
+
+    sr, data = audio_input
+    sf.write(file=audio_filepath, data=data, samplerate=sr)
+
+    return (
+        f"""✅ Updated {dataset_dir}/text.csv \n✅ Saved recording to {audio_filepath}""",
+        None,
+    )
+
+
+def setup_gradio_demo():
+    custom_css = ".gradio-container { max-width: 450px; margin: 0 auto; }"
+    with gr.Blocks(css=custom_css) as demo:
+        gr.Markdown(
+            """
+            # 🎤 Speech-to-text Dataset Recorder
+            ### 1. Select your language
+            """
+        )
+        selected_lang = gr.Dropdown(
+            choices=list(languages_dict.keys()), value="", label="Select a language"
+        )
+        gr.Markdown(
+            "### 2. Click the Common Voice tab to use text data from the Common Voice dataset "
+            "**_OR_** Click the Custom data to write your own text data"
+        )
+        with gr.Row():
+            with gr.Tab("Common Voice"):
+                gr.Markdown("### 3. Click **Load language text dataset**.")
+                load_lang_button = gr.Button("Load Common Voice text dataset")
+                dataset_loaded = gr.Markdown()
+
+                gr.Markdown(
+                    "### 4. Set an index and click **Load from index** or use **← Previous** / **Next →** to navigate sentences."
+                )
+                with gr.Row():
+                    with gr.Column(scale=1):
+                        index = gr.Number(value=0, label="Skip to index")
+                    with gr.Column(scale=3):
+                        load_index_button = gr.Button("Load from index")
+                        with gr.Row():
+                            previous_sentence_button = gr.Button("← Previous")
+                            next_sentence_button = gr.Button("Next →")
+
+                cv_sentence_textbox = gr.Text(
+                    label="Read and record the following sentence"
+                )
+                gr.Markdown(
+                    "_Note: Make sure the recording is not longer than 30 seconds._"
+                )
+                cv_audio_input = gr.Audio(
```

Comment:
Is it possible to play the audio you recorded?

![image](https://github.com/user-attachments/assets/40f55b24-f1f8-44ce-84ab-5834c41b28e6)

Reply:
I have downloaded the `rec_X.wav` files locally but they seem to be empty, not sure what is going on

Reply:
oh! why is your view like that? :sweat_smile: yes that player can the option to play the audio, but its cut off because of that weird rendering... something is going wrong with gradio I guess, cause also my `.wav` files are fine

Reply:
not sure why your saved audio files dont work... didnt manage to reproduce even with the latest `5.15.0` gradio

Reply:
allright I might try again on Monday. I tested did on a fresh Codespaces where I followed the installation commands on the description 🤷